### PR TITLE
ticketmananager improvements - closes #160

### LINKED ISSUE #160: Improvements for the Ticket Manager
I suggest adding new ticket closing flag: **rejected** to use with any ticket which was sent by mistake, don't contain enough information (and sender is not providing it) to resolve it. This flag could also be used when the issue is not related to any part of the achievement (image, text or code). For example sometimes players send tickets because achievement didn't register on the site because of connection problems on one of the sides.

### PR DESCRIPTION
**Pull request by meleu**
Implementing improvements mentioned in #160.

## Allowing tickets to be reopened

Option available for devs, admins and the reporter. Also, automatically add a comment when a ticket is reopened and who reopened it.

## Adding more ways to close a ticket

Actually this was the original request made by @JuliaWolska in #160. In this PR I'm allowing tickets to be closed (and not exactly being marked as "Resolved") and offering some options for the reason of closing.

Again, any change on the ticket status, automatically add a comment saying what was done and who did it.

## Screenshots

Developer view of an open ticket (in the comments you can notice some comments automatically added when I changed the ticket status):

**EDIT:** changed radio buttons to a dropdown list. [See the comment below](https://github.com/RetroAchievements/RAWeb/pull/167#issuecomment-395936233)

devview

Reporter view of an open ticket:

**EDIT:** changed radio buttons to a dropdown list. [See the comment below](https://github.com/RetroAchievements/RAWeb/pull/167#issuecomment-395936233)

ticketreporterview

Anything more complex than that I would prefer to implement after launching RAWeb 2.0.

### CODE REVIEW

**Comment by kdecks** [comment]
Wrong Rom and Network problems should be "Resolved", not closed.

Also would a drop down be possible?

**Comment by meleu** [comment]
> Wrong Rom and Network problems should be "Resolved", not closed.

The "right" thing to do would be use only two statuses for tickets: `Open` or `Closed` (a `Resolved` ticket is in fact a Closed one), any additional info should be in the comments. And that's what I want to do for RAWeb 2.0.

I just didn't go through this way because it would require more work (changing existent data in the database) and we wan to avoid it for now.

The reasoning I'm using to flag a ticket as `Resolved` is "the dev had to modify something in the achievement". That's why I'm not flagging "wrong rom" and "network problems" as `Resolved`. But, again, in RAWeb 2.0 they will be all marked simply as `Closed`.


> would a drop down be possible?

I'll try to make a mockup tonight and post here for evaluation.

**Comment by kdecks** [comment]
> The "right" thing to do would be use only two statuses for tickets: Open or Closed (a Resolved ticket is in fact a Closed one). Any additional info should be in the comments. That's what I want to do for RAWeb 2.0.

I see your logic. 

If we are going have a list of closing reasons why not put most major cases for ticket closing? It will save time for notes and help us categorize issues.

**Comment by meleu** [comment]
@kdecks Yeah, I agree that it would be pretty handy and also wanna see it on the ticket manager. And after trying some approaches to achieve that, I noticed that the code was starting to become a big mess. I'm sure it will be easier to handle with Laravel.

That being said, I'll leave this PR as it is and keep yours and Salsa's suggestions for the future.

**Comment by meleu** [comment]
@Jamiras

> "add comments below" should apply to all of those states

Some of them automatically add a comment (like in the screenshot below). But I agree, I'll put a text above the dropdown box encouraging the user to add comments.

ticketcomments

> network problems could also just be handled by mistaken report

Agreed. But I made them separate for a reason: I'm providing the "Close - Mistaken report" option only for the reporter (by the way, that's why it appears as the first option on that dropdown, the screenshot was taken by a reporter who is a dev). 

Summing up: the `Mistaken report` option is just a way to let the user quickly say "I opened this ticket unnecessarily, please ignore it.".

> order in the drop down is important.

I tried to categorize like this:

- (1) `Mistaken report` - only appears if you are the reporter
- (1) `Resolved`
- (1) `Demote`
- (4) `Close` options

I'll put the `Close` options alphabetically sorted but letting the `Another reason` as the last one.

File: public/ticketmanager.php [hunk 1]
```diff
@@ -542,10 +543,24 @@
...
                     echo "</tr>";
                 }
 
+                echo "<tr>";
+                echo "<td></td><td colspan='6'>";
+
+                $userAwarded = getUserUnlockAchievement( $reportedBy, $achID, $unlockData );
+                if( $userAwarded )
+                {
+                    echo "$reportedBy earned this achievement at ". getNiceDate( strtotime( $unlockData[0][ 'Date' ] ) );
```
Comment (meleu) [inline public/ticketmanager.php]: @luchaos I'm under the impression that hardcoding the `0` index here looks a bit "inelegant". Am I being too nitpicky here?

**Comment by kdecks** [comment]
perfect
